CIP-0189? | Variable Deposits as Pledge (#1204)
cardano-foundation/CIPs: CIP-XXXX/README.md

---
CIP: ?
Title: Variable Deposits as Pledge
Category: Ledger
Status: Proposed
Authors:
    - Ryan Wiley <rian222@gmail.com>
Implementors: []
Discussions:

>     - https://github.com/cardano-foundation/CIPs/pull/?

Reviewer: ```suggestion
    - Original PR: https://github.com/cardano-foundation/CIPs/pull/1204
```
`this` as usual.

Reviewer: @Cerkoryn I'll commit some of these routine things now that we are getting technically oriented review, to keep the PR thread unobstructed...

Created: 2026-05-31
License: CC-BY-4.0
---

## Abstract

> Cardano currently treats stake pool pledge as a declared value rather than ADA that is directly locked as pledge. This creates operational confusion between "declared pledge" and "live pledge", requires a separate owner-stake satisfaction check, and allows mangled-address constructions where apparent pledge can be inflated through stake credential "frankenwallets". Meanwhile, dReps have only a flat registration deposit and no equivalent pledge primitive, limiting future designs that may want dReps to have visible skin in the game.

Reviewer: ```suggestion
Cardano currently treats stake pool pledge as a declared value rather than ADA that is directly locked as pledge. This creates operational confusion between "declared pledge" and "live pledge", requires a separate owner-stake satisfaction check, and allows mangled-address constructions where apparent pledge can be inflated through stake credential "mangled addresses". Meanwhile, dReps have only a flat registration deposit and no equivalent pledge primitive, limiting future designs that may want dReps to have visible skin in the game.
```
(or _franken addresses_ if you prefer)

This is the main term to correct: TL;DR replacing "Frankenwallet" with "mangled" or "franken address" which I am sure you mean.
* <ins>**_The_**</ins> [Frankenwallet](https://frankenwallet.com) is a procedure for building an air-gap or sandbox OS on a drive that handles only secure & crypto workflow: its name comes from patching together mutually supportive workflow tools & sometimes bits of hardware.
* <ins>**_A_**</ins> [franken address](https://cardano.stackexchange.com/questions/9784/guarantee-not-dispensing-native-assets-to-a-franken-address) is another name for [what's more generally called a _mangled address_](https://cardano.stackexchange.com/questions/tagged/mangled-address) — suggested, as you say, in CIP-0019 (cc @KtorZ) but not defined there.

I coined the Frankenwallet term & published its original outline before any of these dated references to "franken address" but — perhaps because of little overlap between the target audiences — the terms have failed to differentiate perfectly.  This is first time I've seen one term being used in place of the other: but not the first time someone assumed a relationship between the two (which there isn't).

Maybe because of that tendency it might help your CIP if you used the equivalent but distinct term "mangled address" which might also help the two "franken" terms to be used less ambiguously in the long run... although you have author's license to use whichever you think makes the best connection with your proposal's own target audience.

The same mechanism is introduced for dReps. The existing dRep deposit parameter is renamed to `minDrepDeposit` and becomes the minimum allowed dRep pledge deposit. A dRep may lock any amount at or above that minimum as dRep pledge. This CIP does not change dRep voting power formulas or define a dRep compensation scheme. It creates a simple, queryable, on-chain pledge primitive that future CIPs may use for dRep incentives, compensation eligibility, concentration mitigation, etc.

Pledged stake remains economically active. Pool pledge deposits continue to earn staking rewards through the pool they pledge to and continue to count toward the dRep chosen by the pool operator's voting delegation. dRep pledge deposits continue to count toward that dRep's voting power and continue to earn staking rewards through the stake pool chosen by the dRep.

> ## Motivation: why is this CIP necessary?

Reviewer: ```suggestion
## Motivation: Why is this CIP necessary?
```
(1 of 2) All these are capitalised now — pending modifications under way to the template & its nearly-complete validation workflow (cc @Ryun1)

### Ledger queries and ecosystem visibility

Ledger state queries SHOULD expose:

- `poolPledgeDeposit` for each registered pool,
- effective pool pledge for the current and relevant future reward snapshots,
- pledge owner credential for each registered pool pledge deposit,
- `drepPledgeDeposit` for each registered dRep,
- effective dRep pledge for the current and relevant future governance snapshots,
- pledge owner credential for each registered dRep pledge deposit.

Wallets, CLIs, explorers, and governance tools SHOULD display pool pledge and dRep pledge as locked ADA. Tools SHOULD distinguish:

- pledge deposit currently locked,
- pledge effective for the current reward or governance snapshot,
- pending change that will become effective at a future snapshot, if applicable.

Tools SHOULD stop presenting pool pledge as both "declared pledge" and "live pledge". Under this CIP, the pledge deposit is the pledge.

> ## Rationale: how does this CIP achieve its goals?

Reviewer: ```suggestion
## Rationale: How does this CIP achieve its goals?
```
(2 of 2)

This also removes the confusing split between declared pledge and live pledge. If 100,000 ADA is locked as `poolPledgeDeposit`, the pool has 100,000 ADA of pledge.

### Pledged ADA remains economically active

This CIP does not remove pledged ADA from staking rewards or governance voting power. It changes how the ADA is locked and attributed as pledged stake.

Pool pledge deposits are required to support the pool they pledge to for stake-pool accounting, but their governance voting power follows the pledge owner credential's dRep delegation. dRep pledge deposits are required to support the dRep they pledge to for governance accounting, but their staking rewards and stake-pool weight follow the pledge owner credential's stake-pool delegation.

This preserves the expected behavior that locked pledge is still ADA participating in the protocol. Because pledged ADA is held in ledger deposit accounting rather than ordinary UTxOs, the ledger MUST include pledged stake in the relevant stake-pool and governance distributions by explicit deterministic rule.

> ### The frankenwallet path is removed

Reviewer: ```suggestion
### The mangled address path is removed
```
Similarly to the first comment above (https://github.com/cardano-foundation/CIPs/pull/1204#discussion_r3331338790): only asking that the term you choose here (`mangled` or `franken`) matches the one in your Abstract.

Renaming them to `minPoolDeposit` and `minDrepDeposit` makes that meaning explicit while preserving the conceptual role of the existing parameters:

- there is one minimum pool pledge deposit,
- there is one minimum dRep pledge deposit,
- each registered pool or dRep has one actual locked pledge deposit.

This requires documentation, tooling, and constitutional guardrails to be updated so that the renamed parameters are understood as minimums rather than fixed amounts.

### dRep pledge is deliberately limited in this CIP

dRep compensation and dRep voting-power concentration are important topics, but they are not settled by this proposal. This CIP introduces dRep pledge as an on-chain primitive and includes it in ordinary delegated stake accounting, but does not add a special dRep compensation or voting-weight formula.

Future CIPs can use dRep pledge as one input while preserving backward compatibility with the primitive introduced here.

### Relationship to CIP-50 and other pledge proposals

> CIP-50 and related reward-sharing proposals change how pledge affects rewards. This CIP changes what pledge is.

Reviewer: ```suggestion
### Relationship to CIP-0050 and other pledge proposals

CIP-0050 and related reward-sharing proposals change how pledge affects rewards. This CIP changes what pledge is.
```
Matching both your use of `CIP-0019` above and the general editorial recommendation (for the benefit of search engines, AI learning, text search, etc.) that writers always use the 4-digit, `0`- padded CIP numbers in `CIP-` tags.

The two ideas are complementary. A reward formula can still include pledge influence, leverage caps, or other pledge-based behavior. Under this CIP, those formulas read pledge from locked pool deposits rather than from declared pool parameters and owner stake checks.

## Path to Active

### Acceptance Criteria

- [ ] CIP editors confirm that the ledger semantics are complete, unambiguous, and consistent with the CIP process.

> - [ ] Ledger maintainers confirm that pool and dRep deposits can be exposed to the reward calculation, registration rules, and state query interfaces required by this CIP.

Reviewer: ```suggestion
- [x] Ledger reviewers confirm the Ledger part of the specification is complete, unambiguous, and consistent.
- [x] Ledger maintainers confirm that pool and dRep deposits can be exposed to the reward calculation, registration rules, and state query interfaces required by this CIP.
```
The "CIP editors" item here is redundant because:
* the editors would do this with respect to whatever `Category:` a CIP candidate is in, whether Ledger or not... and the CIP process applies to every CIP so such a review is already implicit for all documents merged into the repo;
* completeness & specificity are established _much_ better & more routinely by Ledger reviewers than they are by CIP editors.

If you choose to keep these two items here, they would be a prerequisite for merge: so I've pre-ticked these boxes.  In my opinion, these 2 statements aren't necessary: though we'll tag the usual Ledger reviewers later if you leave these lines in.

   - Update stake and governance distribution calculations so pledged stake remains active for staking rewards and dRep voting power as specified above.

3. Node, CLI, wallet, and query support
   - Implement the ledger changes in cardano-ledger and cardano-node.
   - Add CLI support for registering and updating pools and dReps with explicit pledge deposit values and pledge owner credentials.
   - Add state queries for actual and effective pool/dRep pledge deposits.
   - Update wallet and explorer APIs to display locked pledge, snapshot-effective pledge, and the staking/governance delegations attached to pledge deposits.

4. Testing and rollout
   - Add ledger unit and property tests for registration, updates, refunds, minimum deposit enforcement, reward snapshots, governance snapshots, retirement, and deregistration.
   - Run a dedicated testnet or preview-era test deployment with mainnet-like parameters.
   - File any necessary constitutional or governance actions for parameter guardrail changes.
   - Activate through a hard-fork governance action after ecosystem readiness is confirmed.

## Open questions

> - Should we allow the same locked pledge to support multiple roles, such as SPO and dRep, simultaneously?

Reviewer: This is <ins>a CPS feature</ins> which is out of place as a top level heading in a CIP.  The question itself (and why it's important, even if unanswered) could be posed somehow in your _Rationale_ and maybe some potential answers elaborated based on peer review that develops in this PR.

Author: I'll remove it from the CIP and add it to the OP of the discussion section instead.  I think I had some other open questions that are worth discussing too.

---

> CIP: ?

Reviewer: ```suggestion
CIP: 189
```

This preserves the expected behavior that locked pledge is still ADA participating in the protocol. Because pledged ADA is held in ledger deposit accounting rather than ordinary UTxOs, the ledger MUST include pledged stake in the relevant stake-pool and governance distributions by explicit deterministic rule.

### The mangled address path is removed

CIP-0019 explicitly allows the payment and delegation parts of a Shelley address to be controlled by different entities. This is useful and valid, but it means current pledge satisfaction follows delegation rights rather than payment ownership.

Under this CIP, a pool's pledge is not inferred from payment credentials, stake credentials, or address composition. It is the ADA locked under that pool's `PoolDeposit` purpose. A hybrid address can still exist, and stake can still be delegated normally, but it cannot inflate a pool's pledge unless ADA is actually locked as that pool's pledge deposit.

> This also addresses the pledge-as-a-service concern described in cardano-node [issue 2109](https://github.com/IntersectMBO/cardano-node/issues/2109). A third party may still fund or provide ADA for a pool pledge deposit, but doing so creates a real locked deposit and therefore a materially different trust relationship from merely sharing or registering stake credentials.

Reviewer 2: do we know the impact of this on the network? is this widespread?

Author: @Ryun1 I tried to figure out how that could be researched, but unfortunately I don't know if it can be definitively shown when a pool operator is using a mangled address for pledge.  I think the best effort require a lot of probabilities and guess work.
